src/material/input/testing/input-harness.ts

```diff
@@ -111,5 +111,10 @@ export class MatInputHarness extends MatFormFieldControlHarness {
     if (newValue) {
       await inputEl.sendKeys(newValue);
     }
+
+    // Some input types won't respond to key presses (e.g. `color`) so to be sure that the
+    // value is set, we also set the property after the keyboard sequence. Note that we don't
+    // want to do it before, because it can cause the value to be entered twice.
+    await inputEl.setProperty('value', newValue);
```

Comment:
We should update the method description, right? The goal for this method was to simulate the real input typing (e.g. in e2e tests). I'm wondering if this special logic should be opt-in through a flag, or a separate method? Just throwing out some idea's here.

Reply:
The method still does what is said in the description, but it has a fallback for the input that you can't type in.

Reply:
Yeah, but that should be documented IMO. If the value cannot be set through typing (e.g. `readonly`) then I don't expect it to have the new value.

Reply:
That's actually why I'm thinking whether it would make sense to have this as a separate method (explaining the use-case and downsides)

Reply:
I think that would confuse people more than anything, because we'd have two ways of doing more or less the same. The consumer cares about that the value is set, they probably don't care whether it's through typing or a property.

Reply:
I definitely would care on my end if I'd write an e2e test and want to test whether the input is type-able or not (i.e. if `readonly` works properly). I'm not saying that a separate method is the right and only option, but it _could_ be one. At least, documenting is important IMO.

Maybe @mmalerba has more thoughts on this.

Reply:
I don't think we want to add this functionality at all. The idea of the harnesses is that they interact with your component like a user would. A user doesn't get to set properties. The way I would expect people to do this in their tests is that they test whatever path in their code causes this property to be set, then they can check that it was actually set using the harness

Reply:
That makes it impossible to interact with some elements though. E.g. the dropdown of the `color` input is inside the Shadow DOM and isn't accessible via a selector.

Reply:
That seems like a pretty rare case. I think that should probably be outside the scope of the harness system, and we just give people a way to get the element, e.g.: 
`TestbedHarnessEnvironment.getElement(el: TestElement): Element`

Reply:
I'm fine with exposing the element and have a way of setting native properties directly, but I was under the impression that we want to avoid having people interact with the `TestElement`.

Reply:
Yeah, as a general rule of thumb I don't want to expose `TestElement`s to users unless there's a good reason to do so. I would consider this a good reason though

Reply:
Wouldn't it be a little inconsistent though? Also it assumes that people know the internals of the harness. Since `getValue` is based on the `value` property there are cases where it'll work correctly (e.g. if you initialized the input with `value="foo"`) and others where it won't.

Reply:
I'm open to changing how `getValue` works, but I definitely don't want to add a `setProperty` method to `TestElement`. I think the idea of interacting with components like a user would is pretty core to the way the harnesses were designed and I wouldn't want to start violating that